common/lib/dependabot/git_metadata_fetcher.rb

```diff
@@ -52,6 +53,9 @@ def fetch_upload_pack_for(uri)
       response = fetch_raw_upload_pack_for(uri)
       return response.body if response.status == 200
 
+      response_with_git = fetch_raw_upload_pack_with_git_for(uri)
+      return response_with_git.body if response_with_git.status == 200
```

Comment:
It looks like we chose to stick with `status` and `body` so that the error reporting below (.i.e `Octokit::Error.from_response(response)`) could parse it. But the fields don't match. [source](https://github.com/octokit/octokit.rb/blob/be7c105ec7b70e6dc5f83339df82edb3860a9458/lib/octokit/error.rb#L11-L13).

I'd like to suggest that we either:

1. leave the variable as `response_with_git` and remove the non-essential bits like the suggestion to remove the body fields when the `git ls-remote` process fails... 
2. or rename `response_with_git` to response.

If we chose option 2 then we potentially lose some error reporting from the response that came from `fetch_raw_upload_pack_for`. So I think I prefer option 1. WDYT?

Reply:
I originally had it as 2, but I didn't like how that overrode the rest of the error handling around the HTTP response...

I don't know enough Ruby to know how to aggregate the results of the two responses and their respective structs, so the `raise` "does the right thing" and reports _both_ errors.

I'm down with 1 if that works for you!

Reply:
> I'm down with 1 if that works for you!

That works for me. I can tidy it up if it becomes an issue.

Reply:
Done! Thanks!